flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/SlotInfo.java

```diff
@@ -54,10 +59,27 @@ public interface SlotInfo {
      */
     ResourceProfile getResourceProfile();
 
+    /**
+     * Set the loading.
+     *
+     * @param loadingWeight loading weight to set.
+     */
+    void setLoading(LoadingWeight loadingWeight);
+
+    /** Reset the loading. */
+    void resetLoading();
+
```

Comment:
It's used in the block (org.apache.flink.runtime.jobmaster.slotpool.PendingRequest) by `PhysicalSlot`
`

    boolean fulfill(PhysicalSlot slot) {
        Preconditions.checkState(
                slot.getLoading().equals(getLoading()),
                "Unexpected loading weight, it may be a bug.");
        return slotFuture.complete(slot);
    }
`
So, what about move the method declaration into `PhysicalSlot` ?